src/main/java/com/sap/sgs/phosphor/fosstars/data/github/UsesGithubForDevelopment.java

```diff
@@ -39,31 +56,100 @@ protected Value fetchValueFor(GitHubProject project) throws IOException {
+   * then the method concludes that the projects uses GitHub for development.
    * 
-   * @param project of type {@link GitHubProject}.
-   * @return true if the project is a mirror project. Otherwise, false.
-   * @throws IOException may throw an exception during REST call to GitHub API.
+   * @param repository The project's repository.
+   * @param commits The project's commit history.
+   * @return True if it looks like that the project uses GitHub, false otherwise.
+   *
+   */
+  boolean usesGitHubForDevelopment(
+      GHRepository repository, List<GHCommit> commits, double threshold) {
+
+    List<Predicate<GHRepository>> checks = Arrays.asList(
+
+        // check if the repository doesn't have an explicit link to the original repository
+        r -> StringUtils.isEmpty(r.getMirrorUrl()),
+
+        // check if the description mentions "mirror"
+        r -> !r.getDescription().toLowerCase().contains("mirror"),
+
+        // if GitHub issues are enabled, then it's likely that the project uses GitHub
+        GHRepository::hasIssues,
+
+        // if GitHub pages are enabled, then it's likely that the project uses GitHub
+        GHRepository::hasPages,
+
+        // if GitHub Wiki is enabled, then it's likely that the project uses GitHub
+        GHRepository::hasWiki,
+
+        // if one of the three options for pull requests is enabled, then it looks like
+        // that the project uses pull requests
+        r -> r.isAllowMergeCommit() || r.isAllowRebaseMerge() || r.isAllowSquashMerge(),
+
+        // check if repository is not archived
+        r -> !r.isArchived(),
+
+        // check if the project doesn't have an explicit link to an SVN repository
+        r -> StringUtils.isEmpty(r.getSvnUrl()),
+
+        // check if most of user names in the latest commits are valid GitHub users
+        r -> {
+          if (commits.isEmpty()) {
+            return false;
+          }
+          int resolved = commits.stream()
+              .map(commit -> canResolveUsersFor(commit) ? 1 : 0)
+              .reduce(0, Integer::sum);
+          return resolved / commits.size() >= RESOLVED_USERS_THRESHOLD;
+        }
+    );
+
+    int points = checks.stream()
+        .map(check -> check.test(repository) ? 1 : 0)
+        .reduce(0, Integer::sum);
+
+    return points / checks.size() >= threshold;
+  }
+
+  /**
+   * Checks if user names from a commit point to existing GitHub users.
+   *
+   * @param commit The commit to be checked.
+   * @return True if user names from the commit can be resolved to existing GitHub users,
+   *         false otherwise.
    */
-  private boolean isMirror(GitHubProject project) throws IOException {
-    GHRepository repository = gitHubDataFetcher().repositoryFor(project, github);
-    return StringUtils.isNotEmpty(repository.getMirrorUrl());
+  private static boolean canResolveUsersFor(GHCommit commit) {
+    try {
+      commit.getCommitter();
```

Comment:
I think we need a null check here and in the next line. What are we checking here is not clear to me.

Reply:
If a project doesn't use GitHub as the main development repository, then user ids in the commit messages are not necessary point to GitHub user ids. The `getCommitter()` and `getAuthor()` methods throw an i/o exception if they can't find a GitHub user for a username from the commit message.

I'll add a null check although it I am not sure if the GitHub data fercher can return nulls in the commit list. Anyway, it's better to have a check.